Tools/LyTestTools/ly_test_tools/o3de/multi_test_framework.py

```diff
@@ -0,0 +1,1383 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Simplified O3DE test-writing utilities.
+
+Supports different options for running tests using this framework:
+SingleTest: A single test that runs 1 test in 1 executable/application.
+SharedTest: Multiple tests that run multiple tests in multiple executables/applications.
+BatchedTest: Multiple tests that run 1 executable/application with multiple tests in that 1 executable/application.
+ParallelTest: Multiple tests that run multiple executables/applications with 1 test in each executable/application.
+
+It is recommended that new modules are created with objects that inherit from objects here for new tests.
+Example: creating a new EditorTestSuite(AbstractTestSuite) class for testing the Editor application.
+"""
+
+from __future__ import annotations
+__test__ = False  # Avoid pytest collection & warnings since this module is for test functions, but not a test itself.
+
+import abc
+import functools
+import json
+import inspect
+import logging
+import math
+import os
+import re
+import tempfile
+import threading
+import types
+import warnings
+
+import pytest
+import _pytest.python
+import _pytest.outcomes
+from _pytest.skipping import pytest_runtest_setup as skip_pytest_runtest_setup
+
+import ly_test_tools.environment.process_utils as process_utils
+import ly_test_tools.o3de.editor_test_utils as editor_utils
+from ly_test_tools._internal.managers.workspace import AbstractWorkspaceManager
+from ly_test_tools.o3de.asset_processor import AssetProcessor
+from ly_test_tools.launchers import launcher_helper
+from ly_test_tools.launchers.exceptions import WaitTimeoutError
+from ly_test_tools.launchers.platforms.win.launcher import WinEditor, WinMaterialEditor
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractTestBase(object):
+    """
+    Abstract base Test class
+    """
+    # Maximum time for run, in seconds.
+    timeout = 180
+    # Test file that this test will run.
+    test_module = None
+    # Attach debugger when running the test, useful for debugging crashes. This should never be True on production.
+    # It's also recommended to switch to SingleTest for debugging in isolation.
+    attach_debugger = False
+    # Wait until a debugger is attached at the startup of the test, this is another way of debugging.
+    wait_for_debugger = False
+
+
+class SingleTest(AbstractTestBase):
+    """
+    Test that will be run alone in one executable, with no parallel or serially batched executables.
+    This should only be used if SharedTest, ParallelTest, or BatchedTest cannot be utilized for the test steps.
+    It may also be used for debugging test issues in isolation.
+    """
```

Comment:
Lines 66 to 122 appear to be unused as code in  `editor_test.py` and `material_editor_test.py` all inherit directly from AbstractTestBase. Should be safe to remove, would be misleading to leave in, and may be hiding issues in the `_run_XYZ()` methods below.

Reply:
If the framework functionally depends on classes such as SingleTest, then the child classes would need to inherit from them.

Reply:
If these classes are retained, they should probably be marked abstract and/or protected as they are not intended for direct use.

Reply:
classes inside `MaterialEditorTestSuite` and `EditorTestSuite` will inherit from `SingleTest` and `SharedTest` instead of `AbstractTestBase`